docs/tutorials/reward_networks.rst

```diff
@@ -0,0 +1,131 @@
+    rng = np.random.default_rng(0)
+    from gym.spaces import Box
+    obs_space = Box(np.ones(2), np.ones(2))
+    action_space = Box(np.ones(5), np.ones(5))
+
+.. testcode::
+    :skipif: skip_doctests
+
+    from imitation.util import util
+    from imitation.rewards.reward_wrapper import RewardVecEnvWrapper
+    from imitation.rewards.reward_nets import BasicRewardNet
+
+    reward_net = BasicRewardNet(obs_space, action_space)
+    venv = util.make_vec_env("Pendulum-v1", n_envs=3, rng=rng)
+    venv = RewardVecEnvWrapper(venv, reward_net.predict_processed)
+
+Reward Network Wrappers
+-----------------------
+
+Imitation learning algorithms should converge to a reward function that will theoretically induce the optimal/soft optimal policy. However, these reward functions may not always be well suited for training RL agents, or we may want to modify them to encourage exploration, for instance.
+
+There are two types of wrapper:
+* :class:`ForwardWrapper <imitation.rewards.reward_nets.ForwardWrapper>` allows for direct modification of the results of the reward network's ``forward`` method. It is used during the learning of the reward network and thus must be differentiable. These wrappers are always applied first and are always active. They are used for applying transformations like potential shaping (see :class:`ShapedRewardNet <imitating.rewards.reward_nets.ShapedRewardNet>`).
+* :class:`PredictProcessedWrapper <imitation.rewards.reward_nets.PredictProcessedWrapper>` modifies the predict_processed call to the reward network. Thus this type of reward network wrapper is designed to only modify the reward when it is being used to train/evaluate a policy but *not* when we are taking gradients on it. Thus it does not have to be differentiable.
+
+The most commonly used is the :class:`NormalizedRewardNet <imitating.rewards.reward_nets.NormalizedRewardNet>` which is a predict procssed wrapper. This class uses a normalization layer to standardize the *output* of the reward function using its running mean and variance, which is useful for stabilizing training. When a reward network is saved, its wrappers are saved along with it, so that the normalization fit during reward learning can be used during future policy learning or evaluation.
+
+.. testcode::
+    :skipif: skip_doctests
+
+    from imitation.rewards.reward_nets import NormalizedRewardNet
+    from imitation.util.networks import RunningNorm
+    train_reward_net = NormalizedRewardNet(
+        reward_net,
+        normalize_output_layer=RunningNorm,
+    )
+
+.. note::
+    The reward normalization wrapper does _not_ function identically to stable baselines3's `VecNormalize <https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html#stable_baselines3.common.vec_env.VecNormalize>`_ environment wrapper. First, it does not normalize the observations. Second it normalizes the reward based on the reward networks's mean and variance on and *not* a running estimate of the return.
+
```

Comment:
This note confuses me. You seem to say that the NromalizedRwardNet Wrapper on a running estimate but then what you describe below *does* sound like a running estimate.

Also I think the last sentence might have one "on" too many.